modules/imgproc/src/sumpixels.simd.hpp

```diff
@@ -180,61 +481,852 @@ struct Integral_SIMD<uchar, float, double>
+                    el4h_2 = v_cvt_f32(el4hi_2) + el4l_2;
+                    el4h_3 = v_cvt_f32(el4hi_3) + el4l_3;
+                    el4h_4 = v_cvt_f32(el4hi_4) + el4l_4;
+                    prev_1 = v_broadcast_element<v_float32::nlanes - 1>(el4h_1);
+                    prev_2 = v_broadcast_element<v_float32::nlanes - 1>(el4h_2);
+                    prev_3 = v_broadcast_element<v_float32::nlanes - 1>(el4h_3);
+                    prev_4 = v_broadcast_element<v_float32::nlanes - 1>(el4h_4);
+#endif
+                    v_float32 el4_1, el4_2, el4_3, el4_4, el4_5, el4_6, el4_7, el4_8;
+                    v_zip(el4l_1, el4l_3, el4_1, el4_2);
+                    v_zip(el4l_2, el4l_4, el4_3, el4_4);
+                    v_zip(el4h_1, el4h_3, el4_5, el4_6);
+                    v_zip(el4h_2, el4h_4, el4_7, el4_8);
+                    v_zip(el4_1, el4_3, el4l_1, el4l_2);
+                    v_zip(el4_2, el4_4, el4l_3, el4l_4);
+                    v_zip(el4_5, el4_7, el4h_1, el4h_2);
+                    v_zip(el4_6, el4_8, el4h_3, el4h_4);
+                    v_store(sum_row + j                        , el4l_1 + vx_load(prev_sum_row + j                        ));
+                    v_store(sum_row + j + v_float32::nlanes    , el4l_2 + vx_load(prev_sum_row + j + v_float32::nlanes    ));
+                    v_store(sum_row + j + v_float32::nlanes * 2, el4l_3 + vx_load(prev_sum_row + j + v_float32::nlanes * 2));
+                    v_store(sum_row + j + v_float32::nlanes * 3, el4l_4 + vx_load(prev_sum_row + j + v_float32::nlanes * 3));
+                    v_store(sum_row + j + v_float32::nlanes * 4, el4h_1 + vx_load(prev_sum_row + j + v_float32::nlanes * 4));
+                    v_store(sum_row + j + v_float32::nlanes * 5, el4h_2 + vx_load(prev_sum_row + j + v_float32::nlanes * 5));
+                    v_store(sum_row + j + v_float32::nlanes * 6, el4h_3 + vx_load(prev_sum_row + j + v_float32::nlanes * 6));
+                    v_store(sum_row + j + v_float32::nlanes * 7, el4h_4 + vx_load(prev_sum_row + j + v_float32::nlanes * 7));
+                }
+
+                for (float v4 = sum_row[j - 1] - prev_sum_row[j - 1],
+                           v3 = sum_row[j - 2] - prev_sum_row[j - 2],
+                           v2 = sum_row[j - 3] - prev_sum_row[j - 3],
+                           v1 = sum_row[j - 4] - prev_sum_row[j - 4]; j < width; j += 4)
+                {
+                    sum_row[j]     = (v1 += src_row[j])     + prev_sum_row[j];
+                    sum_row[j + 1] = (v2 += src_row[j + 1]) + prev_sum_row[j + 1];
+                    sum_row[j + 2] = (v3 += src_row[j + 2]) + prev_sum_row[j + 2];
+                    sum_row[j + 3] = (v4 += src_row[j + 3]) + prev_sum_row[j + 3];
+                }
+            }
+        }
+        else
+        {
+            return false;
+        }
+        vx_cleanup();
+
+        return true;
+    }
+};
+
+#if CV_SIMD128_64F && !CV_AVX512_SKX
```

Comment:
Why is excluded `CV_AVX512_SKX`?
Do we want `CV_SIMD_WIDTH <= 32` here instead?

Reply:
There is already a AVX512 version for doubles.  See in above code.